library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/GooglePlayServicesLocationProvider.java

```diff
@@ -121,15 +126,16 @@ public void onConnectionSuspended(int i) {
             suspendedConnectionIteration++;
             getSourceProvider().connectGoogleApiClient();
         } else {
-            LogUtils.logI("GoogleApiClient connection is suspended, calling fail...");
-            failed(FailType.GOOGLE_PLAY_SERVICES_CONNECTION_FAIL);
+            LogUtils.logI("GoogleApiClient connection is suspended, start DefaultLocation ");
+            sourceListener.onConnectionSuspended(i);
```

Comment:
here now, you are also violating the configuration. See the if block above, it checks whether configuration says it should fail or not, and in this case it should but you're switching back to defaultProviders.
Maybe you can even create another configuration part called "switchToDefaultsOnFail" in GooglePlayServicesConfiguration and implement that way further.